src/brogue/Items.c

```diff
@@ -2517,6 +2517,20 @@ void itemDetails(char *buf, item *theItem) {
                                     abs(theItem->enchant1 + 1));
                             strcat(buf, buf2);
                             break;
+                        case RING_AWARENESS:
+                            if (theItem->enchant1 > -1) {
+                                sprintf(buf2, "\n\nWith this ring equipped, you will have a %i%% chance to automatically notice adjacent traps and secret doors. (If the ring is enchanted, the chance will %s %i%%.) ",
+                                        min(100, max(0, theItem->enchant1 * 20 + 20)),
```

Comment:
Perhaps we should consider factoring this out, with the stealth calculation, to use a single function

Reply:
Also there's the fact that secret doors are 75% as discoverable as traps. We don't want to tell the player they have 100% chance to discover an adjacent door when that's not true. Maybe if we say you will have "around a X% chance" where X is real value times 0.875?

Reply:
oh, thanks. i gotta do a big editing pass on this. appreciate all the info and ideas!